### PR TITLE
Small fixes

### LINKED ISSUE #39: Read on mangadex but not on MAL shows wrong next
I have some manga that I started reading before a MAL entry was added. Now there is one available but it's not yet on my list, so it marks the first chapter as next, even though lastMangaDexChapter is 2.

### PR DESCRIPTION
**Pull request by cpiber**
Found a few small things when testing for #27 

When no lastMyAnimeListChapter is found, use lastMangaDexChapter (would show 1st as next), fixes #39
updateManga force should always send a request
don't update on comments page (was legacy reader)

### CODE REVIEW

File: scripts/myMangaDex.js [hunk 1]
```diff
@@ -202,7 +201,7 @@ class MyMangaDex {
 				let { requestURL, body } = buildMyAnimeListBody(usePepper, this.manga, this.csrf, setStatus);
 
 				// Send the POST request to update the manga if there as a change
-				if (doUpdate) {
+				if (force || doUpdate) {
```
Comment (cpiber) [inline scripts/myMangaDex.js:204]: A small but crucial fix, this prevented things like QuickAdd

File: scripts/myMangaDex.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ class MyMangaDex {
 
 		// Choose page
 		if ((this.pageUrl.indexOf(urls.follows) > -1 && this.pageUrl.indexOf("/manga/") == -1) ||
-			(this.pageUrl.indexOf(urls.group) > -1 && (this.pageUrl.indexOf("/chapters/") > -1 || (this.pageUrl.indexOf("/manga/") == -1 && this.pageUrl.indexOf("/comments/") == -1))) ||
+			(this.pageUrl.indexOf(urls.group) > -1 && (this.pageUrl.indexOf("/chapters/") > -1 || (this.pageUrl.indexOf("/manga/") == -1 && this.pageUrl.indexOf("/comments") == -1))) ||
```
Comment (Glagan) [inline scripts/myMangaDex.js:46]: I don't think this is necessary ? Did the ``/comments`` page trigerred the extension ?
This conditions makes it so only ``/chapters`` validate the condition, and both ``/chapters`` and ``/comments`` can't be in the URL at the same time.

Comment (cpiber) [reply scripts/myMangaDex.js:46]: I got the update notification on the comments page, adding this stopped it

Comment (Glagan) [reply scripts/myMangaDex.js:46]: Sorry I misread the change, yes there is an useless ``/`` char.

**Comment by Glagan** [comment]
Any more issues or I can merge and publish an update ?

**Comment by cpiber** [comment]
I think there's nothing else. Though I think I'll work on #41 today/tomorrow, if you want to wait before publishing

**Comment by cpiber** [comment]
Since you didn't merge yet I decided to start pushing here instead of opening a new PR

File: scripts/myMangaDex.js [hunk 3]
```diff
@@ -128,144 +128,135 @@ class MyMangaDex {
...
-							body: body,
-							redirect: "follow",
-							credentials: "include",
-							headers: {
-								"Content-Type": "application/x-www-form-urlencoded",
-								"accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8"
-							}
-						}
-					});
+				} else if (this.manga.lastMyAnimeListChapter >= 0 && doMyAnimeList &&
```
Comment (cpiber) [inline scripts/myMangaDex.js:239]: Currently only suppressing *updated*, waiting for your input

**Comment by Glagan** [comment]
I don't think you should remove any notifications, they all make sense even if there is no MyAnimeList ID.  
Even the started one, since if ``updateMD`` (or something like this) is enabled it will set the **MangaDex** status to **Reading**.

**Comment by Glagan** [comment]
If the ``updateMDList`` option is enabled the **MangaDexList** status is not updated if there is no **MyAnimeList** ID.

**Comment by cpiber** [comment]
> I don't think you should remove any notifications, they all make sense even if there is no MyAnimeList ID.
> Even the started one, since if `updateMD` (or something like this) is enabled it will set the **MangaDex** status to **Reading**.

So notifications are good as they are?

> If the `updateMDList` option is enabled the **MangaDexList** status is not updated if there is no **MyAnimeList** ID.

Because status is fetched from MAL... Do you say we always set it to follow if no MAL ID is set? Or should we find a way to find if it is the last chapter and if so set it to completed? Should be possible since MangaDex has then END-badge (if a power-uploader decides to set it)

**Comment by Glagan** [comment]
Yes, but your removed the "Updated" one if there is no MyAnimeList ID you should put it back.

I use the MyAnimeList status sometimes only, I always set it to *Reading* by default and updates only if it's the last chapter (if it's set on MyAnimeList).
But I use MyAnimeList to know if it was already set and I don't check using MangaDex, you're right, so it would set it to *Reading* every chapter... I don't know, there is a function to check if the title is already in your MangaDexList, we could use that to know if it's already set to *Reading* or not if the option is enabled, and do that only when ``firstRun`` is set to true.

**Comment by cpiber** [comment]
I made it so that the list is only updated, if the title is not on any list, would you also update if dropped etc?

**Comment by cpiber** [comment]
I'm currently tracking a new bug, where setting the score in the modal on the title page doesn't update it. logging reveals that the manga property is updated but right after the loop it is set back, no idea why. Didn't test other options. I'll work on it tomorrow but if you have any ideas they'd be much appreciated.

File: scripts/myMangaDex.js [hunk 4]
```diff
@@ -128,144 +128,154 @@ class MyMangaDex {
...
-								"Content-Type": "application/x-www-form-urlencoded",
-								"accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8"
-							}
-						}
-					});
+		const doUpdate = doMyAnimeList && this.manga.lastMyAnimeListChapter != realChapter;
+		const oldStatus = this.manga.status;
+		/*if (this.mangaDexScore > 0) {
+			this.manga.score = this.mangaDexScore;
+		}*/
```
Comment (cpiber) [inline scripts/myMangaDex.js:211]: This always overrides the updated score from the modal. it's not possible for this block to be executed in any other situation I believe (mangaDexScore on other pages not set)
Maybe we can find a better place for it?

Comment (Glagan) [reply scripts/myMangaDex.js:211]: I don't really know if I was really using the MangaDex score for anything.

**Comment by cpiber** [comment]
I think this should be everything, I didn't find anything else.

Also concerning the changes proposed by Jerrk regarding the follows page, what do you think of them and what were your ideas? I'll start working on them in the next few days, should I use this PR or open a new one?

**Comment by Glagan** [comment]
Open a new one since it's not related to this one.

**Pull request merged**